crates/superposition_types/src/config.rs

```diff
@@ -0,0 +1,274 @@
+            log::error!("Condition validation error: Context is empty");
+            return Err("Context should not be empty".to_owned());
+        }
+        jsonlogic::expression::Expression::from_json(&json!(condition_map)).map_err(
+            |msg| {
+                log::error!("Condition validation error: {}", msg);
+                msg
+            },
+        )?;
+        Ok(Self(condition_map))
+    }
+
+    fn validate_data_for_exp(condition_map: Map<String, Value>) -> Result<Self, String> {
+        let condition_val = json!(condition_map);
+        let ast = jsonlogic::expression::Expression::from_json(&condition_val).map_err(
+            |msg| {
+                log::error!("Condition validation error: {}", msg);
+                msg
+            },
+        )?;
+        let dimensions = ast.get_variable_names().map_err(|msg| {
+            log::error!("Error while parsing variable names : {}", msg);
+            msg
+        })?;
+        if dimensions.contains("variantIds") {
+            log::error!("experiment's context should not contain variantIds dimension");
+            return Err(
+                "experiment's context should not contain variantIds dimension"
+                    .to_string(),
+            );
+        }
+        Ok(Self(condition_map))
+    }
+}
+
+impl_try_from_map!(Cac, Condition, Condition::validate_data_for_cac);
+impl_try_from_map!(Exp, Condition, Condition::validate_data_for_exp);
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct Context {
+    pub id: String,
+    pub condition: Condition,
+    pub priority: i32,
+    pub override_with_keys: [String; 1],
+}
+
+impl Contextual for Context {
+    fn get_condition(&self) -> Condition {
+        self.condition.clone()
+    }
+}
+
+#[repr(C)]
+#[derive(Serialize, Deserialize, Clone, Debug, Default)]
+pub struct Config {
+    pub contexts: Vec<Context>,
+    pub overrides: HashMap<String, Overrides>,
+    pub default_configs: Map<String, Value>,
+}
+
+impl Config {
+    pub fn filter_by_dimensions(&self, dimension_data: &Map<String, Value>) -> Self {
+        let filtered_context =
+            Context::filter_by_eval(self.contexts.clone(), dimension_data);
+
+        let filtered_overrides: HashMap<String, Overrides> = filtered_context
+            .iter()
+            .flat_map(|ele| {
+                let override_with_key = &ele.override_with_keys[0];
+                self.overrides
+                    .get(override_with_key)
+                    .map(|value| (override_with_key.to_string(), value.clone()))
+            })
+            .collect();
+
+        Self {
+            contexts: filtered_context,
+            overrides: filtered_overrides,
+            default_configs: self.default_configs.clone(),
+        }
+    }
+
+    pub fn filter_default_by_prefix(
+        &self,
+        prefix_list: &HashSet<String>,
+    ) -> Map<String, Value> {
+        filter_keys_by_prefix(self.default_configs.clone(), prefix_list)
+    }
```

Comment:
just a wrapper to filter out default config specifically, as I wanted to keep a single code for `filter_config_keys_by_prefix` and is not provided openly (available in crate only), so that if anyone wants to use that function outside, can introduce proper constructs via some Trait or some function implementation specific to the type of data needing it